pkg/apis/pipeline/v1/result_validation.go

```diff
@@ -66,3 +65,47 @@ func validateObjectResult(tr TaskResult) (errs *apis.FieldError) {
 	}
 	return nil
 }
+
+// validateValue validates the value of the TaskResult.
+// It requires that enable-step-actions is true, the value is of type string
+// and format $(steps.<stepName>.results.<resultName>)
+func (tr TaskResult) validateValue(ctx context.Context) (errs *apis.FieldError) {
+	if tr.Value != nil {
+		if !config.FromContextOrDefaults(ctx).FeatureFlags.EnableStepActions {
+			return apis.ErrGeneric("feature flag %s should be set to true to fetch Results from Steps using StepActions.", config.EnableStepActions)
+		}
+		if tr.Value.Type != ParamTypeString {
+			return &apis.FieldError{
+				Message: fmt.Sprintf(
+					"Invalid Type. Wanted string but got: \"%v\"", tr.Value.Type),
+				Paths: []string{
+					fmt.Sprintf("%s.type", tr.Name),
+				},
+			}
+		}
+		if tr.Value.StringVal != "" {
+			_, _, err := ExtractStepResultName(tr.Value.StringVal)
+			if err != nil {
+				return &apis.FieldError{
+					Message: fmt.Sprintf("Could not extract step and result name from TaskResult. Expect value to look like $(steps.<stepName>.results.<resultName>) but got %v. Got error: %v", tr.Value.StringVal, err),
+					Paths: []string{
+						fmt.Sprintf("%s.value", tr.Name),
+					},
+				}
+			}
+		}
+	}
+	return nil
+}
+
+// ExtractStepResultName extracts the step name and result name from a string matching
+// formtat $(steps.<stepName>.results.<resultName>).
+// If a match is not found, an error is retured.
+func ExtractStepResultName(value string) (string, string, error) {
+	re := regexp.MustCompile(`\$\(steps\.(.*?)\.results\.(.*?)\)`)
+	rs := re.FindStringSubmatch(value)
+	if len(rs) != 3 {
+		return "", "", fmt.Errorf("Did not find two matches.")
+	}
+	return rs[1], rs[2], nil
```

Comment:
are we going to use the returned values in following PRs? Looks like they are not used for now.

Reply:
yes we are.